lib/srv/uacc/uacc.h

```diff
@@ -235,4 +235,37 @@ static int uacc_has_entry_with_user(const char *utmp_path, const char *user) {
     return errno == 0 ? UACC_UTMP_ENTRY_DOES_NOT_EXIST : errno;
 }
 
+// Low level C function to add a new entry to the failed login log.
+// This function does not perform any argument validation.
```

Comment:
What does `This function does not perform any argument validation.` mean?

Validation is critical in C code. And I see you validating `btmp_path` below.

Reply:
I'm not actually sure, it's just how the other C functions are in this file. In practice everything should be validated on the Go side, but we can add validation here too if you want.

Reply:
Generally I decided to write as much logic in Go here for simplicity and have C just trust what Go passes in. Otherwise we end up with a lot more C code that could've been cleaner Go.

The only requirement is that we document what properties the arguments must uphold clearly.